src/app/components/lesson-card/LessonCardComponent.tsx

```diff
@@ -0,0 +1,114 @@
+import Hls from "hls.js";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { unlockedStatus } from "./constants";
+import { ILesson } from "../../services";
+import { Title, Content, Video } from "./styles";
+import { LockOutlined } from "@mui/icons-material";
+import { useDispatch } from "react-redux";
+import { coursePageSlice, useAppSelector } from "../../redux";
+import { Colors } from "../../common/assets";
+
+export const LessonCardComponent: React.FC<ILesson> = ({
+  id,
+  link,
+  order,
+  previewImageLink,
+  status,
+  title,
+}) => {
+  const [showVideo, setShowVideo] = useState(false);
+  const [downloadVideo, setDownloadVideo] = useState(false);
+  const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const { courseId } = useAppSelector(
+    (state) => state.persistedReducer.coursesListPage
+  );
+  const { coursesProgress } = useAppSelector(
+    (state) => state.persistedReducer.coursePage
+  );
+  const { setLessonProgress } = coursePageSlice.actions;
+
+  useEffect(() => {
+    if (showVideo) {
+      var video = document.getElementById(`video-${order}`);
```

Comment:
The same idea as in previous recommendation: **`useRef`** hook is a good choice for this option.

Reply:
Thanks! Done.